### PR TITLE
Do not reset trial state once a license has been issued.

### PR DESCRIPTION
**Pull request by pebrc**
Fixes https://github.com/elastic/cloud-on-k8s/issues/3659

Addresses an issue observed when the trial status secret would not be found due to an outdated client cache and the trial controller would re-initialise the trial state from scratch invalidating an already issued license.

This PR checks therefore for `trialState.IsEmpty` instead of `trialState.IsTrialStarted`.

### CODE REVIEW

File: pkg/controller/license/trial/trial_controller.go [hunk 1]
```diff
@@ -179,11 +179,16 @@ func (r *ReconcileTrials) startTrialActivation() error {
...
 func (r *ReconcileTrials) completeTrialActivation(license types.NamespacedName) (reconcile.Result, error) {
 	if r.trialState.CompleteTrialActivation() {
-		status, err := licensing.ExpectedTrialStatus(r.operatorNamespace, license, r.trialState)
+		expectedStatus, err := licensing.ExpectedTrialStatus(r.operatorNamespace, license, r.trialState)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-		return reconcile.Result{}, r.Update(&status)
+		var existingStatus corev1.Secret
+		if err := r.Get(k8s.ExtractNamespacedName(&expectedStatus), &existingStatus); err != nil {
```
Comment (pebrc) [inline pkg/controller/license/trial/trial_controller.go:187]: This is mainly for the overeager fake client which requires a resource version on every update. The fact that we are hitting this issue only now tells me that none of the other tests executed this code path 🤷

Comment (sebgl) [reply pkg/controller/license/trial/trial_controller.go:187]: How about calling `ReconcileSecret()` instead? Manipulating the resource version here feels a bit weird.

**Review by sebgl — COMMENTED** [review]
LGTM, I left one comment worth addressing imo

**Review by sebgl — APPROVED** [review]
LGTM

**Pull request merged**